src/lib/lessons.ts

```diff
@@ -4,23 +4,49 @@ import getAccessTokenFromCookie from '../utils/get-access-token-from-cookie'
 import {loadLessonComments} from './lesson-comments'
 import {sanityClient} from 'utils/sanity-client'
 import groq from 'groq'
+import some from 'lodash/some'
+import isEmpty from 'lodash/isEmpty'
 
+// next_up_url can be derived on the front-end from collection and
+// staff_notes_url seems to be null for all lessons
+// code_url is only used in a select few Kent C. Dodds lessons
 const lessonQuery = groq`
-*[_type == 'lesson' && slug == $slug][0]{
+*[_type == 'lesson' && slug.current == $slug][0]{
   title,
-  slug,
+  'slug': slug.current,
   description,
+  resource->_type == 'videoResource' => {
+    ...(resource-> {
+      'media_url': hslUrl,
+      'transcript': transcriptBody,
+      'transcript_url': transcriptUrl,
+      duration,
+      'subtitles_url': subtitlesUrl,
+    })
+  },
+  'free_forever': isCommunityResource,
+  'path': '/lessons/' + slug.current,
+  'thumb_url': thumbnailUrl,
+  'repo_url': repoUrl,
+  'code_url': codeUrl,
+  'createdAt': eggheadRailsCreatedAt,
+  'updatedAt': eggheadRailsUpdatedAt,
+  'publishedAt': eggheadRailsPublishedAt,
```

Comment:
In the lesson schema we made explicitly prefixed these field names with `egghead` rails to make it clear that these are values from rails we are tracking. 

We had the thought of using Sanity's field group's feature to separate these fields into their own tab. https://www.sanity.io/docs/field-groups

Reply:
I had to snake_case the left-hand-side names, but otherwise, cree's comments still stand.